staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -31,147 +31,161 @@ import (
 )
 
 // NestedFieldCopy returns a deep copy of the value of a nested field.
-// false is returned if the value is missing.
-// nil, true is returned for a nil field.
-func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{}, bool) {
-	val, ok := nestedFieldNoCopy(obj, fields...)
-	if !ok {
-		return nil, false
+// An error is returned if the value is missing.
+// No error  is returned for a nil field.
+func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{}, error) {
```

Comment:
the alternative for all the signature changes: `(interface{}, bool, error)`. That would be more like map access `value, found, err := NestedFieldCopy(...)`

Reply:
@ash2k we have to understand whether the field is missing or eixsts with wrong type. Wdyt?

Reply:
@sttts I think I like `(interface{}, bool, error)` return signature more than what we have today and more than what is here in this PR right now `(interface{}, error)`. "field not found" is not an error, should be a bool.

Reply:
I wonder why I didn't do it myself...

Reply:
Done.